### DISCUSSION TITLE
How to add shortcut to Form?

### DISCUSSION THREAD

**Discussion by heejit** [cogentcore/core#1249] [General]
How can I add shortcut to the form/Body so event handler call before calling it to the child widget.

i try following but not worked:
```
this.Body.Styler(func (s *styles.Style) {
		s.SetAbilities(true, abilities.Focusable)
	})
	this.Body.OnKeyChord(func (e events.Event) {
		fmt.Println(e.KeyChord())
		fmt.Println(e.Modifiers())
	})
```

**Comment by heejit** [comment]
Found it ->
```
	this.Body.OnFirst(events.KeyChord, func (e events.Event) {
		if e.KeyChord() == "Control+S" {
			this.SaveNote()
		}
	})
```

**Reply by kkoreilly** [reply]
Yes, `OnFirst` is one approach that works. You could also make a `Button` and do `SetShortcut` on it, which will do that automatically (see the [buttons page](https://www.cogentcore.org/core/widgets/basic/buttons)). Either approach works, and you can do whichever you prefer.

**Reply by heejit** [reply]
I need few more shortcut keys so OnFirst approach is working fine.
